linux/devdoc/async_socket_linux_requirements.md

```diff
@@ -77,52 +77,16 @@ MOCKABLE_FUNCTION(, ASYNC_SOCKET_HANDLE, async_socket_create, EXECUTION_ENGINE_H
 
 **SRS_ASYNC_SOCKET_LINUX_11_001: [** `async_socket_create` shall allocate a new async socket and on success shall return a non-`NULL` handle. **]**
 
-**SRS_ASYNC_SOCKET_LINUX_11_002: [** If `execution_engine` is NULL, `async_socket_create` shall fail and return `NULL`. **]**
+**SRS_ASYNC_SOCKET_LINUX_11_002: [** `execution_engine` shall be allowed to be `NULL`. **]**
 
 **SRS_ASYNC_SOCKET_LINUX_11_003: [** If `socket_handle` is `INVALID_SOCKET`, `async_socket_create` shall fail and return `NULL`. **]**
 
-**SRS_ASYNC_SOCKET_LINUX_11_004: [** `async_socket_create` shall increment the reference count on `execution_engine`. **]**
+**SRS_ASYNC_SOCKET_LINUX_11_005: [** `async_socket_create` shall retreive an `COMPLETION_PORT_HANDLE` object by calling `platform_get_completion_port`. **]**
 
-**SRS_ASYNC_SOCKET_LINUX_11_005: [** `async_socket_create` shall initialize the global thread. **]**
+**SRS_ASYNC_SOCKET_LINUX_11_101: [** `async_socket_create` shall increment the reference count of the `COMPLETION_PORT_HANDLE` object by calling completion_port_inc_ref. **]**
```

Comment:
These 2 requirements are contradictory with themselves.

If these `COMPLETION_PORT_HANDLE` are in any way ref-counted, and the factory/getter for them seems to be "platform_get_completion_port" then there's no need to inc_ref them, the factory will return them already inc_ref'd.

Wondering how the tests didn't leak...

Reply:
Ok, I was wondering about that.  In our base code we have it both ways.  I first started it with the increment happening in the platform call, then as I looked at example I changed it.  I can change it back.